lib/class-wp-rest-blocks-renderer-controller.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Blocks Renderer REST API: WP_REST_Blocks_Renderer_Controller class
+ *
+ * @package gutenberg
+ * @since ?
+ */
+
+/**
+ * Controller which provides REST endpoint for rendering blocks.
+ *
+ * @since ?
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Blocks_Renderer_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructs the controller.
+	 *
+	 * @access public
+	 */
+	public function __construct() {
+
+		// @codingStandardsIgnoreLine - PHPCS mistakes $this->namespace for the namespace keyword.
+		$this->namespace = 'gutenberg/v1';
+		$this->rest_base = 'blocks-renderer';
+	}
+
+	/**
+	 * Registers the necessary REST API routes.
+	 *
+	 * @access public
+	 */
+	public function register_routes() {
+
+		// @codingStandardsIgnoreLine - PHPCS mistakes $this->namespace for the namespace keyword.
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<name>[\w-]+\/[\w-]+)', array(
```

Comment:
One idea: instead of letting `name` be a parameter, we could register over all of the registered blocks and create an endpoint for each one. This could then allow for the block's attributes schema to be used as an `attributes` object arg in the `args` here, and it would eliminate the need for `prepare_attributes` logic in `get_item_output`. (See https://core.trac.wordpress.org/ticket/38583)

This would allow for a list of all the server-side renderable blocks to be discovered by looking at the REST API schema. Only the blocks returned by `get_dynamic_block_names()` would be included.